Bugzilla.pm

```diff
@@ -767,6 +768,23 @@ sub elastic {
     $class->process_cache->{elastic} //= Bugzilla::Elastic->new();
 }
 
+sub check_rate_limit {
+    my ($class, $name, $id) = @_;
+    my $params = Bugzilla->params;
+    if ($params->{rate_limit_active}) {
+        my $rules = decode_json($params->{rate_limit_rules});
```

Comment:
It isn't allowed to be invalid (no way of saving an invalid value) although maybe covering that is a good idea.